pkg/agent/memberlist/cluster.go

```diff
@@ -472,13 +481,31 @@ func (c *Cluster) ShouldSelectIP(ip, externalIPPool string) (bool, error) {
 	if !ok {
 		return false, fmt.Errorf("local Node consistentHashMap has not synced, ExternalIPPool %s", externalIPPool)
 	}
-	node := consistentHash.Get(ip)
-	if node == "" {
-		klog.Warningf("No valid Node chosen for IP %s in externalIPPool %s", ip, externalIPPool)
+	node := consistentHash.GetWithFilters(ip, fillters...)
```

Comment:
@jianjuns Since the implementation is changed, agents to update and agent that have been updated will select different Nodes for an Egress when upgrading. Do you think it's ok given the feature is still alpha?

Reply:
It is not ideal, but should not be a big matter for an alpha feature. We should document this in the release note.